static-site/app/blog/utils.ts

```diff
@@ -0,0 +1,72 @@
+import fs from "fs";
+import matter from "gray-matter";
+import path from "path";
+import { fileURLToPath } from "url";
+
+import parseMarkdown from "./parseMarkdown";
+
+export interface BlogPost {
+  author: string;
+  blogUrlName: string;
+  date: string;
+  mdstring: string;
+  sidebarTitle: string;
+  title: string;
+}
+
+// Scans the ./static-site/content/blog directory for .md files and
+// returns a chronologically-sorted array of posts, each with some
+// basic metadata and the raw (unsanitized) markdown contents.
+export function getBlogPosts(): BlogPost[] {
```

Comment:
There are two functions that have duplicated JS and TS versions:

- getBlogPosts
    - static-site/src/util/blogPosts.js
    - static-site/app/blog/utils.ts
- parseMarkdown
    - static-site/app/blog/parseMarkdown.ts
    - static-site/src/util/parseMarkdown.js

I understand that the JS versions are still used by `scripts/generate-blog-feeds.js`. Is there a plan to deduplicate these at some point?

Reply:
I could convert `scripts/generate-blog-feeds` to TS — which would allow the old `getBlogPosts()` to be removed — but it felt out of scope of the issue.

The old `parseMarkdown()` is also used in `static-site/src/components/sourceInfoHeading.jsx`, so that needs both community and groups (at least) to be converted over before it can be removed.

Reply:
Thanks for the explanation. I [still think](https://github.com/nextstrain/nextstrain.org/pull/1032#discussion_r1833206146) it'd be useful to add a small comment at the "deprecated" JS file indicating that it's been ported over to TS. Otherwise I will keep asking for clarification about these duplicates in PR reviews.

Reply:
Added deprecation notices to these two files.